src/translate.py

```diff
@@ -2306,7 +2440,20 @@ def deref(
 
 
 def is_trivial_expression(expr: Expression) -> bool:
-    # Determine whether an expression should be evaluated only once or not.
+    """Compute whether an EvalOnceExpr should be marked as `trivial = True`."""
+    # NaivePhiExpr could be made trivial, but it's better to keep it symmetric
+    # with PlannedPhiExpr to avoid different deduplication between passes,
+    # since that can cause naive phis to end up in the final output.
+    if isinstance(expr, (Literal, GlobalSymbol, SecondF64Half)):
+        return True
+    if isinstance(expr, AddressOf):
+        return all(is_trivial_expression(e) for e in expr.dependencies())
+    return False
+
+
+def should_wrap_transparently(expr: Expression) -> bool:
+    """Compute whether an EvalOnceExpr should be marked as `transparent = True`,
+    thus not using a variable despite multiple uses (unless forced)."""
```

Comment:
Maybe this function should start with `if is_trivial_expression(expr): return True` to enforce that all trivial expressions are transparent? (mostly to avoid having the two diverge?)

Reply:
I'm inclined to keep this as is, honestly, I think it's more readable this way and I don't think there's a practical risk they'll diverge